contracts/core/cross_chain_manager/logic/EthCrossChainManager.sol

```diff
@@ -11,12 +11,26 @@ import "./../interface/IEthCrossChainData.sol";
 contract EthCrossChainManager is IEthCrossChainManager, UpgradableECCM {
     using SafeMath for uint256;
 
+    mapping(address => bool) public whiteListContract;
+    mapping(bytes => bool) public unsetEpochPkBytes;
+
     event InitGenesisBlockEvent(uint256 height, bytes rawHeader);
     event ChangeBookKeeperEvent(uint256 height, bytes rawHeader);
     event CrossChainEvent(address indexed sender, bytes txId, address proxyOrAssetContract, uint64 toChainId, bytes toContract, bytes rawdata);
     event VerifyHeaderAndExecuteTxEvent(uint64 fromChainID, bytes toContract, bytes crossChainTxHash, bytes fromChainTxHash);
-    constructor(address _eccd) UpgradableECCM(_eccd) public {}
+    constructor(address _eccd, uint64 _chainId, address[] memory contracts, bytes memory curEpochPkBytes) UpgradableECCM(_eccd,_chainId) public {
+        for (uint i=0;i<contracts.length;i++) {
+            whiteListContract[contracts[i]] = true;
```

Comment:
Should you have a method to add more lockproxies to the whitelist too? instead of only in the constructor

Reply:
when CCM is deployed , its ownership will be transferred to CCMP , and CCMP only do pause or upgrade , authority management is not easy for addWhiteList function.